dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -111,11 +112,12 @@ internal class RumSessionScope(
         event: RumRawEvent,
         actualWriter: DataWriter<RumEvent>
     ) {
-        if (event is RumRawEvent.AddError ||
+        val isValidBackgroundEvent = event is RumRawEvent.AddError ||
             event is RumRawEvent.AddLongTask ||
             event is RumRawEvent.StartAction ||
             event is RumRawEvent.StartResource
-        ) {
+
+        if (isValidBackgroundEvent && backgroundTrackingEnabled) {
```

Comment:
I'd say if you check first if `backgroundTrackingEnabled` in most of the cases you will not have to compute the `isValidBackgroundEvent`

Reply:
Do you mean `if (backgroundTrackingEnabled && isValidBackgroundEvent)` ? I'd rather  keep it all without nested blocks to have a single else statement

Reply:
yes if you check first the `backgroundTrackingEnabeld` then you will not have to compute the `isValidBackgroundEvent` in case this is `false`

Reply:
```
if(!backgroundTrackingEnabled){
  return
}
val isValidBackgroundEvent = ...
```

Reply:
Yes but that will mean duplicating the else block. And the performance impact of this check seems totally negligible